src/foundry/client/pixi/placeables/primary-canvas-objects/primary-sprite-mesh.d.mts

```diff
@@ -0,0 +1,110 @@
+import type { InexactPartial } from "../../../../../types/utils.d.mts";
+
+// TODO: Remove when #2570 is completed
+declare const PrimaryBaseSamplerShader: BaseSamplerShader;
+
+// TODO: Remove when #2878 is completed
+type TextureAlphaData = unknown;
+
+declare global {
+  /**
+   * A basic PCO sprite mesh which is handling occlusion and depth.
+   */
+  class PrimarySpriteMesh extends PrimaryOccludableObjectMixin(SpriteMesh) {
+    constructor(
+      options?:
+        | InexactPartial<{
+            /** Texture passed to the SpriteMesh. */
+            texture: PIXI.Texture;
+
+            /** The shader class used to render this sprite. */
+            shaderClass: typeof PrimaryBaseSamplerShader;
+
+            /** The name of this sprite. */
+            name: string;
+          }>
+        | PIXI.Texture,
+      shaderClass?: typeof PrimaryBaseSamplerShader,
+    );
+
+    /**
+     * The texture alpha data.
+     */
+    _textureAlphaData: TextureAlphaData | null;
+
+    /**
+     * The texture alpha threshold used for point containment tests.
+     * If set to a value larger than 0, the texture alpha data is
+     * extracted from the texture at 25% resolution.
+     */
+    textureAlphaThreshold: number;
+
+    override _onTextureUpdate(): void;
+
+    override setShaderClass(shaderClass: typeof AbstractBaseShader): void;
```

Comment:
Foundry actually adds the check:
```js
    if ( !foundry.utils.isSubclass(shaderClass, PrimaryBaseSamplerShader) ) {
      throw new Error(`${shaderClass.name} in not a subclass of PrimaryBaseSamplerShader`);
    }
```

So `typeof PrimaryBaseSamplerShader` is more accurate. And actually it's `PrimaryBaseSamplerShader.AnyConstructor` for the same reasons as before.